### PR TITLE
Update IORT SMMUv3 IORT node flags for HTTU to include dirty state support

### LINKED ISSUE #498: [Bug]: SMMUv3 IORT HTTU Override flag needs to be expanded
### Is there an existing issue for this?

- [X] I have searched existing issues

### Current Behavior

The flag for HTTU override in an SMMUv3 node in the IORT table is currently define in MU_BASECORE/MdePkg/Include/IndustryStandard/IoRemappingTable.h:63 as:
```
#define EFI_ACPI_IORT_SMMUv3_FLAG_HTTU_OVERRIDE     BIT1
```

### Expected Behavior

The length of this field is actually 2 bits. Possible values are:

0b0000: Hardware update of the Access flag and dirty state are not supported.
0b0001: Support for hardware update of the Access flag for Block and Page descriptors.
0b0010: As 0b0001, and adds support for hardware update of the Access flag for Block and Page descriptors. Hardware update of dirty state is supported.

For more info see:
- Arm System Memory Management Unit Architecture Specification
- Hardware Updates to Access Flag and Dirty State in the Armv8.1-A architecture 

### Steps To Reproduce

N/A

### Build Environment

```markdown
N/A
```

### Version Information

```text
v2023020002.1.4
```

### Urgency

Low

### Are you going to fix this?

I will fix it

### Do you need maintainer feedback?

Maintainer feedback requested

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by JoeLopez333**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description
Fixes https://github.com/microsoft/mu_basecore/issues/498

The flag for HTTU override in an SMMUv3 node in the IORT table is currently define in MU_BASECORE/MdePkg/Include/IndustryStandard/IoRemappingTable.h:63 as:
```
#define EFI_ACPI_IORT_SMMUv3_FLAG_HTTU_OVERRIDE     BIT1
```

The length of this field is actually 2 bits. Possible values are:

0b0000: Hardware update of the Access flag and dirty state are not supported.
0b0001: Support for hardware update of the Access flag for Block and Page descriptors.
0b0010: As 0b0001, and adds support for hardware update of the Access flag for Block and Page descriptors. Hardware update of dirty state is supported.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested on Arm64 platform

## Integration Instructions

update flags in IORT table if consuming this flag definition

### CODE REVIEW

**Review by os-d — APPROVED** [review]

File: MdePkg/Include/IndustryStandard/IoRemappingTable.h [hunk 1]
```diff
@@ -59,10 +59,13 @@
 #define EFI_ACPI_IORT_SMMUv1v2_INT_FLAG_LEVEL  0x0
 #define EFI_ACPI_IORT_SMMUv1v2_INT_FLAG_EDGE   0x1
 
-#define EFI_ACPI_IORT_SMMUv3_FLAG_COHAC_OVERRIDE    BIT0
-#define EFI_ACPI_IORT_SMMUv3_FLAG_HTTU_OVERRIDE     BIT1
-#define EFI_ACPI_IORT_SMMUv3_FLAG_PROXIMITY_DOMAIN  BIT3
-#define EFI_ACPI_IORT_SMMUv3_FLAG_DEVICEID_VALID    BIT4
+// MU_CHANGE Start
+#define EFI_ACPI_IORT_SMMUv3_FLAG_COHAC_OVERRIDE        BIT0
+#define EFI_ACPI_IORT_SMMUv3_FLAG_HTTU_OVERRIDE_AF      BIT1  // HW update of Access Flag supported
```
Comment (kuqin12) [inline MdePkg/Include/IndustryStandard/IoRemappingTable.h]: Renaming this define might lead to breaking change (depending on whether platforms are consuming this). Do we really need to make the "_AF" into the name?

**Comment by kuqin12** [comment]
Please add "Fixes https://github.com/microsoft/mu_basecore/issues/498" to the PR description.

**Review by kuqin12 — APPROVED** [review]

File: MdePkg/Include/IndustryStandard/IoRemappingTable.h (hunk 1 above)
Comment (JoeLopez333) [reply MdePkg/Include/IndustryStandard/IoRemappingTable.h]: @kuqin12 we could instead do:

```
#define EFI_ACPI_IORT_SMMUv3_FLAG_HTTU_OVERRIDE         BIT1  // HW update of Access Flag supported
#define EFI_ACPI_IORT_SMMUv3_FLAG_HTTU_OVERRIDE_DS   BIT2  // HW update of Access Flag + Dirty Flag supported
```

So the "Access Flag" portion of the override is implied. The comment next to the define should be clear enough

Comment (kuqin12) [reply MdePkg/Include/IndustryStandard/IoRemappingTable.h]: Sure, that would work. Thanks!

**Pull request merged**